src/ngraph_encapsulate_impl.h

```diff
@@ -0,0 +1,112 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#ifndef NGRAPH_TF_ENCAPSULATE_IMPL_H_
+#define NGRAPH_TF_ENCAPSULATE_IMPL_H_
+
+#include <ostream>
+#include <vector>
+
+#include "ngraph/ngraph.hpp"
+
+#include "ngraph_freshness_tracker.h"
+
+#include "ngraph_log.h"
+
+#include "tensorflow/core/framework/tensor_shape.h"
+#include "tensorflow/core/graph/graph.h"
+
+namespace tensorflow {
+
+using NgFunctionIOCache = std::unordered_map<
+    std::shared_ptr<ngraph::runtime::Executable>,
+    std::vector<std::pair<void*, shared_ptr<ng::runtime::Tensor>>>>;
+
+namespace ngraph_bridge {
+
+class NGraphEncapsulateImpl {
+ public:
+  NGraphEncapsulateImpl(string name);
+  Status ComputeSignature(std::vector<Tensor>& input_tensors,
+                          std::vector<TensorShape>& input_shapes,
+                          std::vector<const Tensor*>& static_input_map,
+                          std::stringstream& signature_ss);
+  Status GetNgExecutable(std::vector<Tensor>& input_tensors,
+                         const std::pair<string, int64> ctx_params,
+                         std::vector<TensorShape>& input_shapes,
+                         std::vector<const Tensor*>& static_input_map,
+                         ng::runtime::Backend*& op_backend,
+                         std::shared_ptr<ngraph::runtime::Executable>& ng_exec);
+  Status AllocateNGInputTensors(
+      const std::vector<Tensor>& input_tensors,
+      std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
+      std::vector<TensorShape>& input_shapes, ng::runtime::Backend* op_backend,
+      vector<shared_ptr<ng::runtime::Tensor>>& ng_inputs);
+  Status AllocateNGOutputTensors(
+      std::vector<Tensor*>& output_tensors,
+      std::vector<ng::element::Type> expected_output_types,
+      const std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
+      std::vector<TensorShape>& input_shapes, ng::runtime::Backend* op_backend,
+      vector<shared_ptr<ng::runtime::Tensor>>& ng_outputs,
+      std::vector<std::pair<void*, std::shared_ptr<ng::runtime::Tensor>>>&
+          output_caches);
+
+  // TF Graph for the cluster
+  Graph m_graph;
+  std::unordered_map<std::string, std::shared_ptr<ngraph::runtime::Executable>>
+      m_ng_exec_map;
+  std::unordered_map<std::shared_ptr<ngraph::runtime::Executable>,
+                     std::shared_ptr<ngraph::Function>>
+      m_ng_function_map;
+
+  NgFunctionIOCache m_ng_exec_input_cache_map;
+  NgFunctionIOCache m_ng_exec_output_cache_map;
+
+  int number_of_copies = 0;
```

Comment:
Organize the variables nicely and initialize as much as possible.
(Keep the variables together and functions together)